pages/dashboards/views/multidisease_serology.py

```diff
@@ -0,0 +1,118 @@
+
+# Explicit column orders for deterministic display
+KTH_HEADERS: List[str] = ["Virus type", "Variant", "Protein", "Details", "Host"]
+EXTERNAL_HEADERS: List[str] = ["Pathogen", "Variant", "Protein", "Details", "Host"]
+
+logger = logging.getLogger(__name__)
+
+
+class MultiDiseaseSerology(BaseTemplateView):
+    """Render the multidisease serology dashboard.
+
+    The view performs three lightweight steps on every request: download the two
+    XLSX blobs, parse the first worksheet of each with Polars, and normalise the
+    tables into ordered rows so the template can render simple loops.
+
+    Attributes:
+        template_name: Template that renders the descriptive copy and tables.
+        title: Page title that appears in the shared layout.
+    """
+
+    template_name = "dashboards/multidisease_serology.html"
+    title = "Multi-disease serology"
+
+    def _read_excel_frame(self, url: str) -> pl.DataFrame:
+        """Fetch an Excel sheet using urllib and parse it with Polars."""
+        request = Request(url, headers={"User-Agent": USER_AGENT})
+        with urlopen(request, timeout=REQUEST_TIMEOUT_SECONDS) as response:
+            payload = BytesIO(response.read())
+        payload.seek(0)
+        # Polars read_excel uses Rust calamine engine which requires fastexcel for performance.
+        # This dependency is not installed by default, so it was added manually.
+        # TODO: Remove it once we have a proper data pipeline.
+        frame = pl.read_excel(source=payload, sheet_id=0)
+        # Polars returns a dict when the backend loads multiple worksheets; grab the first sheet.
+        return next(iter(frame.values())) if isinstance(frame, dict) else frame
+
+    def _frame_to_rows(self, frame: pl.DataFrame, headers: List[str]) -> List[List[Any]]:
+        """Project a Polars frame onto expected columns and normalise values."""
+        if frame.is_empty():
+            return []
+
+        projected = frame.select(
+            [
+                (pl.col(header).cast(pl.String, strict=False).fill_null("").alias(header))
+                if header in frame.columns
+                else pl.lit("").alias(header)
+                for header in headers
+            ]
+        )
+        return [list(row) for row in projected.rows()]
+
+    def _load_rows(self, url: str, headers: List[str], source: str) -> List[List[Any]]:
+        """Helper to fetch, parse, and normalise table data."""
+        try:
+            logger.info("serology_fetch_start", extra={"source": source, "url": url})
+            frame = self._read_excel_frame(url)
+            logger.info("serology_fetch_success", extra={"source": source, "records": frame.height})
+        except Exception:
+            logger.exception("serology_fetch_failed", extra={"source": source, "url": url})
+            return []
+        return self._frame_to_rows(frame, headers)
+
+    def get_context_data(self, **kwargs: Any) -> Dict[str, Any]:
```

Comment:
Instead of `BaseTemplateView`, we can use diango's `View` as base class and define `get()` method.

We created the abstract class `BaseTemplateView` to avoid modifying `get_context_data`, for (mostly) all the dashboards we need to do some kinda logic i.e. not the typical "just rendering the template".

So to be in alignment with other dashboard views, we can also do the same here.